lib/elixir/lib/kernel.ex

```diff
@@ -5462,6 +5462,69 @@ defmodule Kernel do
     raise ArgumentError, "modifier must be one of: s, a, c"
   end
 
+  @doc """
+  Handles the `~PID` sigil.
+
+  This function allows creating arbitrary PIDs, even the ones that don't actually exist in the VM.
+  As such, it should only be used for debugging purposes.
+  """
+  @doc since: "1.11.0"
+  def sigil_PID(string, modifiers)
+
+  def sigil_PID(string, []) do
+    :erlang.list_to_pid('<#{string}>')
+  end
+
+  @doc """
+  Handles the `~Port` sigil.
+
+  This function allows creating arbitrary ports, even the ones that don't actually exist in the VM.
+  As such, it should only be used for debugging purposes.
+  """
+  @doc since: "1.11.0"
+  def sigil_Port(string, modifiers)
+
+  def sigil_Port(string, []) do
+    :erlang.list_to_port('~Port<#{string}>')
+  end
+
+  @doc """
+  Handles the `~Reference` sigil.
+
+  This function allows creating arbitrary references, even the ones that don't actually exist in the VM.
+  As such, it should only be used for debugging purposes.
+  """
+  @doc since: "1.11.0"
+  def sigil_Reference(string, modifiers)
+
+  def sigil_Reference(string, []) do
+    :erlang.list_to_ref('#Ref<#{string}>')
+  end
+
+  @doc """
+  Handles the `~URI` sigil.
+
+  See `URI.parse/1` for more information.
+  """
+  @doc since: "1.11.0"
+  defmacro sigil_URI(string, modifiers)
+
+  defmacro sigil_URI({:<<>>, _, [string]}, []) do
+    Macro.escape(URI.parse(string))
+  end
+
+  @doc """
+  Handles the `~Version` sigil.
+
+  See `Version.parse!/1` for more information.
+  """
+  @doc since: "1.11.0"
+  defmacro sigil_Version(string, modifiers)
+
+  defmacro sigil_Version({:<<>>, _, [string]}, []) do
+    Macro.escape(Version.parse!(string))
+  end
```

Comment:
Should we have `~Version` and `~URI` in Kernel? I'd argue they belong in their respective modules so that people have to import them to use them. I don't think they're nearly common enough. Thoughts?

Reply:
This is a good point. On one hand it would be convenient to have them in Kernel but at the same time it would be fairer to have them imported, as libraries outside of Elixir would also require. Does anyone have any further thoughts?